server/controllers/users.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-useless-escape */
+import { Op } from 'sequelize';
+import passport from 'passport';
+import models from '../models';
+import { validateLogin, validateSignup, updateDetails } from '../validations/auth';
+
+const { User } = models;
+
+/**
+ * @exports UserController
+ * @class UserController
+ * @description Handles Social Users
+ * */
+class UserController {
+  /**
+  * Create a new user
+  * @async
+  * @param  {object} req - Request object
+  * @param {object} res - Response object
+  * @param {object} next The next middleware
+  * @return {json} Returns json object
+  * @static
+  */
+  static async create(req, res, next) {
+    try {
+      const { error } = validateSignup(req.body);
+      if (error !== null) {
+        const errorValue = error.details[0].message.replace(/\"/g, '');
+        return res.status(400).json({ status: 400, error: errorValue });
+      }
+      const { username, email, password } = req.body;
+      const validUsername = await User.findOne({
```

Comment:
instead of doing this here,I guess you could use your `findOrCreate` with `email or username`.

additionally, you could also just use the `User.create` then check the error it revels in the catch block and (if it a unique violation) return the `email already taken` or `username already taken` error (this would usually be in `err.error`) or if it's not a unique violation then you `next(err)`